### ISSUE TITLE
False erorrs with generic interceptors

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#70311]
**Version Used**: 
```
Microsoft Visual Studio Professional 2022
Version 17.8.0 Preview 2.0
VisualStudio.17.Preview/17.8.0-pre.2.0+34112.27
Microsoft .NET Framework
Version 4.8.09037
```

**Steps to Reproduce**:

Compile the following code:
```cs
#nullable enable
using System.Runtime.CompilerServices;

class C
{
  public string? Method1<T>(T arg) => null;
}

static class Program
{
  public static void Main()
  {
    var c = new C();
    string? x = null;

    c.Method1(x);
  }
}

static class D
{
  [InterceptsLocation("Program.cs", 16, 7)] // error
  public static string? Generic<T>(this C s, T arg) => arg?.ToString();
}

namespace System.Runtime.CompilerServices
{
  [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
  public class InterceptsLocationAttribute : Attribute
  {
    public InterceptsLocationAttribute(string path, int line, int character) { }
  }
}
```

**Expected Behavior**:
No errors. The intercepted method and the interceptor method have the same signatures

**Actual Behavior**:
Error `CS9148: Interceptor must have a 'this' parameter matching parameter 'C this' on 'C.Method1<string>(string)'.` is reported.

The error doesn't make sense, as the interceptor already has `this` parameter of the correct type
Moreover, if you change it to non-generic version the error disappears, so it seems that the problem is not in the `this` parameter:
```cs
static class D
{
  [InterceptsLocation("Program.cs", 16, 7)] // error
  public static string? NonGeneric(this C s, string? arg) => arg?.ToString();
}
```
